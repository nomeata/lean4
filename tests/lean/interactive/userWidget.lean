import Lean
open Lean

@[widget_module]
def helloWidget : Widget.Module where
  javascript := "
    import * as React from 'react';
    export default function(props) {
      const name = props.name || 'world'
      return React.createElement('p', {}, name + '!')
    }"

#widget helloWidget
  --^ widgets

structure HelloWidgetProps where
  name? : Option String := none
  deriving Server.RpcEncodable

#widget helloWidget with { name? := "lean" : HelloWidgetProps }
  --^ widgets

show_panel_widgets [helloWidget with { name? := "global" : HelloWidgetProps }]

section
show_panel_widgets [local helloWidget with { name? := "local" : HelloWidgetProps }]
--^ widgets
end

namespace Foo
show_panel_widgets [scoped helloWidget with { name? := "scoped" : HelloWidgetProps }]
end Foo

open scoped Foo
--^ widgets

show_panel_widgets [-helloWidget]
--^ widgets
