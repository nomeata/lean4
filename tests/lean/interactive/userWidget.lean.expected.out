{"widgets":
 [{"range":
   {"start": {"line": 12, "character": 0},
    "end": {"line": 12, "character": 19}},
   "props": {},
   "javascriptHash": "2874772427606622741",
   "id": "helloWidget"}]}
{"sourcetext":
 "\n    import * as React from 'react';\n    export default function(props) {\n      const name = props.name || 'world'\n      return React.createElement('p', {}, name + '!')\n    }"}
{"widgets":
 [{"range":
   {"start": {"line": 19, "character": 0},
    "end": {"line": 19, "character": 63}},
   "props": {"name": "lean"},
   "javascriptHash": "2874772427606622741",
   "id": "helloWidget"}]}
{"sourcetext":
 "\n    import * as React from 'react';\n    export default function(props) {\n      const name = props.name || 'world'\n      return React.createElement('p', {}, name + '!')\n    }"}
{"widgets":
 [{"range": null,
   "props": {"name": "local"},
   "javascriptHash": "2874772427606622741",
   "id": "helloWidget"},
  {"range": null,
   "props": {"name": "global"},
   "javascriptHash": "2874772427606622741",
   "id": "helloWidget"}]}
{"sourcetext":
 "\n    import * as React from 'react';\n    export default function(props) {\n      const name = props.name || 'world'\n      return React.createElement('p', {}, name + '!')\n    }"}
{"sourcetext":
 "\n    import * as React from 'react';\n    export default function(props) {\n      const name = props.name || 'world'\n      return React.createElement('p', {}, name + '!')\n    }"}
{"widgets":
 [{"range": null,
   "props": {"name": "scoped"},
   "javascriptHash": "2874772427606622741",
   "id": "helloWidget"},
  {"range": null,
   "props": {"name": "global"},
   "javascriptHash": "2874772427606622741",
   "id": "helloWidget"}]}
{"sourcetext":
 "\n    import * as React from 'react';\n    export default function(props) {\n      const name = props.name || 'world'\n      return React.createElement('p', {}, name + '!')\n    }"}
{"sourcetext":
 "\n    import * as React from 'react';\n    export default function(props) {\n      const name = props.name || 'world'\n      return React.createElement('p', {}, name + '!')\n    }"}
{"widgets": []}
